File: files/200-299/293/CH-293.py
import pandas as pd

path = "200-299/293/CH-293 Advanced Sorting.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows=7)
test = pd.read_excel(path, usecols="F:H", skiprows=1, nrows=7).rename(columns=lambda col: col.replace('.1', ''))

sizes = ["S", "M", "L", "XL", "XXL", "XXXL"]
result = input.sort_values("Size", key=lambda x: x.map({s: i for i, s in enumerate(sizes)})).reset_index(drop=True)

print(result.equals(test)) # True

File: files/200-299/293/CH-293.R
library(tidyverse)
library(readxl)

path = "files/200-299/293/CH-293 Advanced Sorting.xlsx"
input = read_excel(path, range = "B2:D8")
test  = read_excel(path, rsange = "F2:H8")

result = input %>%
  arrange(fct_relevel(Size, "S", "M", "L", "XL", "XXL", "XXXL")) 

all.equal(result, test)
# [1] TRUE